src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/fields/ConsignmentFields.scala

```diff
@@ -5,16 +5,41 @@ import java.util.UUID
 import io.circe.generic.auto._
 import sangria.macros.derive._
 import sangria.marshalling.circe._
-import sangria.schema.{Argument, Field, InputObjectType, LongType, ObjectType, OptionType, fields}
+import sangria.schema.{Argument, Field, InputObjectType, IntType, ObjectType, OptionType, fields}
 import uk.gov.nationalarchives.tdr.api.auth.{ValidateSeries, ValidateUserOwnsConsignment}
-import uk.gov.nationalarchives.tdr.api.graphql.ConsignmentApiContext
 import uk.gov.nationalarchives.tdr.api.graphql.fields.FieldTypes._
+import uk.gov.nationalarchives.tdr.api.graphql.{ConsignmentApiContext, DeferFileProgress, DeferTotalFiles}
 
 object ConsignmentFields {
   case class Consignment(consignmentid: Option[UUID] = None, userid: UUID, seriesid: UUID)
   case class AddConsignmentInput(seriesid: UUID)
+  case class AntivirusProgress(processedFiles: Int)
+  case class FileCheckProgress(antivirusProgress: AntivirusProgress)
+
+  implicit val FileCheckProgressType: ObjectType[Unit, FileCheckProgress] =
+    deriveObjectType[Unit, FileCheckProgress]()
+  implicit val AntivirusProgressType: ObjectType[Unit, AntivirusProgress] =
+    deriveObjectType[Unit, AntivirusProgress]()
+
+  implicit private val ConsignmentType: ObjectType[Unit, Consignment] = ObjectType(
+    "Consignment",
+    fields[Unit, Consignment](
+      Field("consignmentid", OptionType(UuidType), resolve = _.value.consignmentid),
+      Field("userid", UuidType, resolve = _.value.userid),
+      Field("seriesid", UuidType, resolve = _.value.seriesid),
+      Field(
+        "totalFiles",
+        IntType,
+        resolve = context => DeferTotalFiles(context.value.consignmentid)
```

Comment:
This didn't occur to me on Friday, but we could use the GraphQL query argument, which is a `UUID`, rather than the result, which is an `Option[UUID]` so we don't have all the awkward Option handling in DeferTotalFiles and DeferFileProgress.

I think you'd replace `context.value.consignmentid` with `context.args.arg(ConsignmentIdArg)`.

Reply:
We tried this yesterday and it didn't work - it looks like `context.args` contains the arguments for the _field_, not the arguments for the parent query.

It would be good to make `context.value.consignmentid` a non-optional value, but it's a change to the existing API so we'd have to be careful not to break any of the clients.